### PR TITLE
Adjustment in README.md and add test.md and architecture.md 

### PR DESCRIPTION
**Pull request by pablomedeiross**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,49 @@
-# Clean architecture in Go
-This project is a simple sample of an implementation Clean Architecture in GoLang.
+# Clean architecture in Golang
+This project is a sample of an implementation Clean Architecture in GoLang.
```
Comment (vitorfariaz) [inline README.md]:
```suggestion
This project is a sample of an implementation of Clean Architecture in GoLang.
```

File: README.md [hunk 2]
```diff
@@ -1,30 +1,49 @@
...
 
 <br/>
 
-## Use case 
-User API : This API is responsible by management users. With it us going to create and search users.
+### Summary
+The clean architecture is a model proposed by [Robert C. Martin (Uncle Bob)](http://cleancoder.com/).
+Briefly, this model seeks to strengthen the following points in our software:
+
+
```
Comment (vitorfariaz) [inline README.md]: Exclude this empty lines

File: README.md [hunk 3]
```diff
@@ -1,30 +1,49 @@
...
+- Increase expressivity of problem that the software is solutioning.
+- Independence of external structures, example: Frameworks, DB, UI, WEB).
```
Comment (vitorfariaz) [inline README.md]:
```suggestion
- Independence of external structures, example: Frameworks, DB, UI, WEB.
```

File: README.md [hunk 4]
```diff
@@ -36,4 +55,4 @@ cd go-clean-architecture
 # run application
 go run main.go 
 
-```
+```
```
Comment (vitorfariaz) [inline README.md]: exclude this

File: README.md (hunk 2 above)
Comment (pablomedeiross) [reply README.md]: empty lines excluded

**Pull request merged**